src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c

```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
+            func_800FE484();
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953F8C.s")
+void func_809543D4(BgIngate* this, GlobalContext* globalCtx) {
+    u8 talkState;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953F9C.s")
+    talkState = Message_GetState(&globalCtx->msgCtx);
+    if (((talkState == 4) || (talkState == 5)) && func_80147624(globalCtx)) {
+        switch (this->dyna.actor.textId) {
+            case 0x9E4:
+                this->dyna.actor.textId = 0x9E5;
+                func_80151938(globalCtx, this->dyna.actor.textId);
+                break;
+            case 0x9E5:
+                if (globalCtx->msgCtx.choiceIndex == 0) {
+                    func_800B7298(globalCtx, &this->dyna.actor, 6);
+                    this->unk160 &= ~0x4;
+                    this->actionFunc = func_809541B8;
+                    func_800FE498();
+                    func_8019F208();
+                } else {
+                    if (this->unk164 != 0) {
+                        this->unk164 = &globalCtx->setupPathList[this->unk164->unk1];
+                    }
+                    func_80953F14(this, globalCtx);
+                    gSaveContext.weekEventReg[0x5A] &= 0xBF;
+                    func_8019F230();
+                }
+                func_801477B4(globalCtx);
+                break;
+            case 0x9E6:
+                if (globalCtx->msgCtx.choiceIndex == 0) {
+                    func_80953EA4(this, globalCtx);
+                    gSaveContext.weekEventReg[0x5A] &= 0xBF;
+                    func_8019F208();
+                } else {
+                    this = this;
+                    func_800B7298(globalCtx, &this->dyna.actor, 6);
+                    this->unk160 &= ~0x4;
+                    this->actionFunc = func_809541B8;
+                    func_800FE498();
+                    func_8019F230();
+                }
+                func_801477B4(globalCtx);
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_809541B8.s")
+void BgIngate_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    BgIngate* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    s32 phi_a2;
+    Vec3s* sp38;
+    Vec3f sp2C;
+    Vec3f sp20;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_809542A0.s")
+    if (BgIngate_findActor(this, globalCtx, ACTORCAT_BG, ACTOR_BG_INGATE) == NULL) {
+        DynaPolyActor_Init(&this->dyna, 3);
+        DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &gSichitaiBoatCol);
+        this->unk160 = 0;
+        this->unk160 |= 0x8;
+        this->unk160 |= 0x10;
+        Actor_SetScale(&this->dyna.actor, 1.0f);
+        this->unk164 = func_8013BB34(globalCtx, BGINGATE_GET_FF(&this->dyna.actor), 0);
+        this->dyna.actor.room = -1;
+        if (gSaveContext.weekEventReg[0x14] & 2) {
+            gSaveContext.weekEventReg[0x5A] &= 0xBF;
```

Comment:
```suggestion
            gSaveContext.weekEventReg[0x5A] &= (u8)~0x40;
```

Reply:
Currently don't see this as changed yet.

Reply:
Hmm, I guess my find and replace didn't catch everything. should be fixed now